src/DssSpell.t.sol

```diff
@@ -43,14 +43,41 @@ contract DssSpellTest is DssSpellTestBase {
 
         // Insert new collateral tests here
         checkIlkIntegration(
-            "WBTC-B",
-            GemJoinAbstract(addr.addr("MCD_JOIN_WBTC_B")),
-            ClipAbstract(addr.addr("MCD_CLIP_WBTC_B")),
+            "WBTC-C",
+            GemJoinAbstract(addr.addr("MCD_JOIN_WBTC_C")),
+            ClipAbstract(addr.addr("MCD_CLIP_WBTC_C")),
             addr.addr("PIP_WBTC"),
             true,
             true,
             false
         );
+        checkPsmIlkIntegration(
```

Comment:
`GUSD` token uses another contract for storing the balances that doesn't have the `balanceOf` interface and `giveTokens` can't be used, even passing the store contract address.
This emerged during the Liq 2.0 migration for `GUSD` in #165 and we checked with Gonza the `balances` slot that should be n. `6`.
A workaround for `checkPsmIlkIntegration` could be the following:
```
        if (_ilk == "PSM-GUSD-A") {
            // GUSD uses another contract for storing the balances
            // which doens't have the balanceOf interface, then giveTokens
            // can't be used as it is (even passing the store contract addr)
            hevm.store(
                address(0xc42B14e49744538e3C239f8ae48A1Eaaf35e68a0),
                keccak256(abi.encode(address(this), uint256(6))),
                bytes32(uint256(amount))
            );
        } else {
                giveTokens(token, amount);
        }
        assertEq(token.balanceOf(address(this)), amount);
 ```

Reply:
Also the `approve` interface is not ERC20 compliant (does not return bool).
This could be used instead:
```
// GUSD approve is not ERC20 compliant (does not return bool)
TokenCompatibleGUSD(address(token)).approve(address(join), amount);
```